src/Talis/Persona/Client/Base.php

```diff
@@ -116,50 +123,113 @@ protected function getLogger()
     }
 
     /**
+     * @return \GuzzleHttp\Client
+     */
+    protected function getHTTPClient()
+    {
+        if ($this->httpClient == null) {
+            $this->httpClient = new Client(
+                $this->config['persona_host']
+            );
+        }
+
+        return $this->httpClient;
+    }
+
+    /**
      * Perform the request according to the $curlOptions
      * @param $curlOptions array options to execute cURL with
      * @param $expectResponse set true if you expect a JSON response with a 200, otherwise expect a 204 no content
      * @return array|null
+     * @throws NotFoundException if the http status was a 404
      * @throws \Exception if response not 200 and valid JSON
      */
-    protected function performRequest(array $curlOptions,$expectResponse=true)
+    protected function performRequest($url, array $opts, $expectResponse=true, $addContentType=true, $parseJson=true)
```

Comment:
This `$parseJson` parameter seems oddly fit for a specific purpose.  Is this because of how the method used to behave?